tensorboard/webapp/hparams/_redux/hparams_data_source.ts

```diff
@@ -0,0 +1,176 @@
+  constructor(private readonly http: TBHttpClient) {}
+
+  private getPrefix(experimentIds: string[]) {
+    return experimentIds.length > 1 ? 'compare' : 'experiment';
+  }
+
+  private formatExperimentIds(experimentIds: string[]) {
+    if (experimentIds.length === 1) {
+      return experimentIds[0];
+    }
+
+    return experimentIds.map((eid) => `${eid}:${eid}`).join(',');
+  }
+
+  fetchExperimentInfo(
+    experimentIds: string[]
+  ): Observable<HparamAndMetricSpec> {
+    const formattedExperimentIds = this.formatExperimentIds(experimentIds);
+    return this.http
+      .post<BackendHparamsExperimentResponse>(
+        `/${this.getPrefix(
+          experimentIds
+        )}/${formattedExperimentIds}/${HPARAMS_HTTP_PATH_PREFIX}/experiment`,
+        {experimentName: formattedExperimentIds},
+        {},
+        'request'
+      )
+      .pipe(
+        map((response) => {
+          return {
+            hparams: response.hparamInfos.map((hparam) => {
+              const feHparam = {
+                ...hparam,
+                domain: getHparamDomain(hparam),
+              };
+
+              delete (feHparam as any).domainInterval;
+              delete (feHparam as any).domainDiscrete;
+
+              return feHparam;
+            }),
+            metrics: response.metricInfos.map((metric) => ({
+              ...metric,
+              tag: metric.name.tag,
+            })),
+          };
+        })
+      );
+  }
+
+  fetchSessionGroups(
+    experimentIds: string[],
+    hparamsAndMetricsSpecs: HparamAndMetricSpec
+  ): Observable<SessionGroup[]> {
+    const formattedExperimentIds = this.formatExperimentIds(experimentIds);
+
+    const colParams: BackendListSessionGroupRequest['colParams'] = [];
+
+    for (const hparam of hparamsAndMetricsSpecs.hparams) {
+      colParams.push({hparam: hparam.name});
+    }
+    for (const mectric of hparamsAndMetricsSpecs.metrics) {
+      colParams.push({
+        metric: mectric.name,
+      });
+    }
+
+    const listSessionRequestParams: BackendListSessionGroupRequest = {
+      experimentName: formattedExperimentIds,
+      allowedStatuses: [
+        RunStatus.STATUS_FAILURE,
+        RunStatus.STATUS_RUNNING,
+        RunStatus.STATUS_SUCCESS,
+        RunStatus.STATUS_UNKNOWN,
+      ],
+      colParams,
+      startIndex: 0,
+      // arbitrary large number so it does not get clipped.
+      sliceSize: 1e6,
+    };
+
+    return this.http
+      .post<BackendListSessionGroupResponse>(
+        `/${this.getPrefix(
+          experimentIds
+        )}/${formattedExperimentIds}/${HPARAMS_HTTP_PATH_PREFIX}/session_groups`,
+        listSessionRequestParams,
+        {},
+        'request'
+      )
+      .pipe(
+        map((response) =>
+          response.sessionGroups.map((sessionGroup) => {
+            sessionGroup.sessions = sessionGroup.sessions.map((session) => {
+              if (experimentIds.length > 1) {
+                const [, ...runName] = session.name.split(' ');
```

Comment:
I think comment explaining the structure of the returned session name would be helpful here.